Error in compiling and calling static library under Windows

### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

When I use the latest version of sfml, compile the static library under Windows, and then use the following statement to call it, the following error occurs.
```
find_package(SFML COMPONENTS system window graphics audio CONFIG REQUIRED)
target_link_libraries(web PRIVATE sfml-system sfml-network sfml-graphics sfml-window sfml-audio)
```
Error:
```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		CMake Error at E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake:42 (set_property):
  set_property could not find TARGET OpenAL.  Perhaps it has not yet been
  created.		E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake	42		

```

### Your Environment

- OS / distro / window manager: Windows
- SFML version: 2.6.2
- Compiler / toolchain: vs2022
- Special compiler / CMake flags: cmake 3.30.1


### Steps to reproduce

1. git clone vcpkg
2. Replaced version 2.6.2.
3. vcpkg install sfml:x64-windows-static
4. Call the generated sfml library.
5.Create a vs cmake project, then call sfml in the project to compile the project, and an error occurs.

```
add_executable (web "web.cpp" "web.h")
find_package(SFML COMPONENTS system window graphics audio CONFIG REQUIRED)
target_link_libraries(web PRIVATE sfml-system sfml-network sfml-graphics sfml-window sfml-audio)
```

### Expected behavior

The compiled result can be called normally.

### Actual behavior

Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		CMake Error at E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake:42 (set_property):
  set_property could not find TARGET OpenAL.  Perhaps it has not yet been
  created.		E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake	42

```
1> CMake generation started for configuration: 'x64-Release'.
1> Command line: "C:\Windows\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="RelWithDebInfo" -DCMAKE_INSTALL_PREFIX:PATH="E:\workspace\web\out\install\x64-Release" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="E:/vcpkg/scripts/buildsystems/vcpkg.cmake" -DVCPKG_TARGET_TRIPLET:STRING="x64-windows-static"   "E:\workspace\web" 2>&1"
1> Working directory: E:\workspace\web\out\build\x64-Release
1> [CMake] -- Selecting Windows SDK version 10.0.26100.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.41.34123.0
1> [CMake] -- The CXX compiler identification is MSVC 19.41.34123.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Found ZLIB: optimized;E:/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;E:/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib (found version "1.3.1")
1> [CMake] -- Found BZip2: optimized;E:/vcpkg/installed/x64-windows-static/lib/bz2.lib;debug;E:/vcpkg/installed/x64-windows-static/debug/lib/bz2d.lib (found version "1.0.8")
1> [CMake] -- Looking for BZ2_bzCompressInit
1> [CMake] -- Looking for BZ2_bzCompressInit - found
1> [CMake] -- Found PNG: optimized;E:/vcpkg/installed/x64-windows-static/lib/libpng16.lib;debug;E:/vcpkg/installed/x64-windows-static/debug/lib/libpng16d.lib (found version "1.6.43")
1> [CMake] -- Found OpenAL: E:/vcpkg/installed/x64-windows-static/debug/lib/OpenAL32.lib
1> [CMake] CMake Error at E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake:42 (set_property):
1> [CMake]   set_property could not find TARGET OpenAL.  Perhaps it has not yet been
1> [CMake]   created.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfigDependencies.cmake:87 (sfml_bind_dependency)
1> [CMake]   E:/vcpkg/installed/x64-windows-static/share/sfml/SFMLConfig.cmake:122 (include)
1> [CMake]   E:/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package)
1> [CMake]   web/CMakeLists.txt:7 (find_package)
1> [CMake] -- Found SFML 2.6.2 in E:/vcpkg/installed/x64-windows-static/share/sfml
1> [CMake] -- Configuring incomplete, errors occurred!
1> 'C:\Windows\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="RelWithDebInfo" -DCMAKE_INSTALL_PREFIX:PATH="E:\workspace\web\out\install\x64-Release" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="E:/vcpkg/scripts/buildsystems/vcpkg.cmake" -DVCPKG_TARGET_TRIPLET:STRING="x64-windows-static"   "E:\workspace\web" 2>&1"' execution failed with error: ''C:\Windows\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="RelWithDebInfo" -DCMAKE_INSTALL_PREFIX:PATH="E:\workspace\web\out\install\x64-Release" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="E:/vcpkg/scripts/buildsystems/vcpkg.cmake" -DVCPKG_TARGET_TRIPLET:STRING="x64-windows-static"   "E:\workspace\web" 2>&1"' returned with exit code: 1'.
```
Looking at the compilation output process, the openal32.lib library was found, but an error occurred in set_property.

--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -442,4 +442,9 @@ function(sfml_export_targets)
                   "${CMAKE_CURRENT_BINARY_DIR}/SFMLConfigVersion.cmake"
             DESTINATION ${config_package_location}
             COMPONENT devel)
+
+    # Install custom openal find module
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/FindOpenAL.cmake"
+            DESTINATION ${config_package_location}/Modules
+            COMPONENT devel)
 endfunction()

--- a/cmake/SFMLConfigDependencies.cmake.in
+++ b/cmake/SFMLConfigDependencies.cmake.in
@@ -75,7 +75,6 @@ if(SFML_STATIC_LIBRARIES)
     # sfml-audio
     list(FIND SFML_FIND_COMPONENTS "audio" FIND_SFML_AUDIO_COMPONENT_INDEX)
     if(FIND_SFML_AUDIO_COMPONENT_INDEX GREATER -1)
-        sfml_bind_dependency(TARGET OpenAL FRIENDLY_NAME "OpenAL" SEARCH_NAMES "OpenAL" "openal" "openal32")
         if (NOT FIND_SFML_OS_IOS)
             sfml_bind_dependency(TARGET VORBIS FRIENDLY_NAME "VorbisFile" SEARCH_NAMES "vorbisfile")
             sfml_bind_dependency(TARGET VORBIS FRIENDLY_NAME "VorbisEnc" SEARCH_NAMES "vorbisenc")
@@ -83,6 +82,12 @@ if(SFML_STATIC_LIBRARIES)
         sfml_bind_dependency(TARGET VORBIS FRIENDLY_NAME "Vorbis" SEARCH_NAMES "vorbis")
         sfml_bind_dependency(TARGET VORBIS FRIENDLY_NAME "Ogg" SEARCH_NAMES "ogg")
         sfml_bind_dependency(TARGET FLAC FRIENDLY_NAME "FLAC" SEARCH_NAMES "FLAC")
+
+        # For OpenAL targets use our custom find module so runtime dependencies are properly propagated
+        list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/Modules/")
+        list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_LIST_DIR}/../..") # For it to find our packaged openAL library
+        set(OPENAL_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/../../include") # Include dir isn't needed but find module will error out if not set
+        find_package(OpenAL REQUIRED)
     endif()
 
     if (FIND_SFML_DEPENDENCIES_NOTFOUND)

